addons/test_mail/tests/test_mail_followers.py

```diff
@@ -417,62 +414,154 @@ def test_auto_subscribe_notify_email(self):
 
         subtype = self.env.ref('mail.mt_comment')
-        res = self.env['mail.followers']._get_recipient_data(test, 'comment', subtype.id, pids=common_partner.ids)
-        partner_notif = [r for r in res if r[0] == common_partner.id]
+        res = self.env['mail.followers']._get_recipient_data(test, 'comment', subtype.id, pids=self.common_partner.ids)
+        partner_notif = [r for r in res if r[0] == self.common_partner.id]
         self.assertEqual(len(partner_notif), 1)
         self.assertEqual(partner_notif[0][3], 'email')
 
-@tagged('post_install', '-at_install')
-class UnlinkedNotificationTest(TestMailCommon):
-    def test_unlinked_notification(self):
-        """
-        Check that we unlink the created user_notification after unlinked the related document
-
-        Post install because we need the registery to be ready to send notification
-        """
-        common_partner = self.env['res.partner'].create({"name": "demo1", "email": "demo1@test.com"})
-        user_1 = self.env['res.users'].create({'login': 'demo1', 'partner_id': common_partner.id, 'notification_type': 'inbox'})
-
-        test = self.env['mail.test.track'].create({"name": "Test Track", "user_id": user_1.id})
-        test_id = test.id
+    @users('employee')
+    def test_notification_unlink(self):
+        """ Check that we unlink the created user_notification after unlinked the
+        related document. Post install because we need the registry to be ready
+        to send notifications """
+        test = self.env['mail.test.track'].create({"name": "Test Track", "user_id": self.user_1.id})
         mail_message = self.env['mail.message'].search([
-             ('res_id', '=', test_id),
-             ('model', '=', 'mail.test.track'),
-             ('message_type', '=', 'user_notification')
+            ('res_id', '=', test.id),
+            ('model', '=', 'mail.test.track'),
+            ('message_type', '=', 'user_notification')
         ])
         self.assertEqual(len(mail_message), 1)
         test.unlink()
         mail_message = self.env['mail.message'].search([
-             ('res_id', '=', test_id),
-             ('model', '=', 'mail.test.track'),
-             ('message_type', '=', 'user_notification')
+            ('res_id', '=', test.id),
+            ('model', '=', 'mail.test.track'),
+            ('message_type', '=', 'user_notification')
         ])
         self.assertEqual(len(mail_message), 0)
+
+    @users('employee')
+    def test_recipients_fetch(self):
+        """ Test internals of ``_get_recipient_data`` to ease its maintenance.
+        Even if it is low-level knowing what it produces allows to write short
+        tests. """
+        test_records = self.env['mail.test.simple'].create([
+            {'email_from': 'ignasse@example.com',
+             'name': 'Test %s' % idx,
+            } for idx in range(5)
+        ])
+        # make followers listen to notes to use it against portal
```

Comment:
"against portal" ? Not sure what you mean by that
Is it that everyone will follow notes except portal, so you can test that it's not included in recipients data for notes?